SurfaceTopography/Support/Deprecation.py

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright 2021 Lars Pastewka
+#
+# ### MIT license
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+import warnings
+from functools import wraps
+
+
+def deprecated(version=None, alternative=None):
```

Comment:
Just an idea: Like the original "deprecated" one could add a sentence at the end of the doc string
about the deprecation, the same string as the warning when using it.

Like this our Sphinx documentation would show what is deprecated.

Reply:
I am not sure I know how to do that. I tried adding something to `func.__doc__`, but that does not show up in the help string.

Reply:
Hm... The package "deprecated" seems to do it in a similar way: https://github.com/tantale/deprecated/blob/09c31517c5f864ab07819eb4ec2235671ad43854/deprecated/sphinx.py#L120

I've tried it like that and it seems to work, I'll push it to this branch.

Reply:
Let me know what you think - either we remove it or simply close this conversation issue.

Reply:
Just another comment for starting a review (just understood how this works here).